CMakeLists.txt

```diff
@@ -79,6 +79,17 @@ endif()
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
   ament_lint_auto_find_test_dependencies()
+
+  find_package(performance_test_fixture REQUIRED)
+  # Give cppcheck hints about macro definitions coming from outside this package
+  get_target_property(ament_cmake_cppcheck_ADDITIONAL_INCLUDE_DIRS
+    performance_test_fixture::performance_test_fixture INTERFACE_INCLUDE_DIRECTORIES)
+
+  add_performance_test(benchmark_log benchmark/benchmark_log.cpp)
+  if(TARGET benchmark_log)
+    target_include_directories(benchmark_log PUBLIC include ${console_bridge_INCLUDE_DIRS})
+    target_link_libraries(benchmark_log ${console_bridge_LIBRARIES})
```

Comment:
The `${console_bridge_*}` variables aren't set at configure time when we don't find a system package for `console_bridge` and are about to build it from source (but haven't yet). So this won't work right in that case. This is pretty hard to debug, because once the external package gets built, even if building the test fails during that first build, subsequent builds will find it just fine. Also, even if we force the package to build `console_bridge` from source, if the system package is available, we'll still end up seeing it when we compile the test.

This is the best way I can find to work around the issue. I tried to make it as stable as possible, but I had to make assumptions about `console_bridge`'s use of `GNUInstallDirs` to predict where the output artifacts will end up.

```patch
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,7 +13,12 @@ if(NOT FORCE_BUILD_VENDOR_PKG)
 endif()
 
 macro(build_console_bridge)
-  set(extra_cmake_args)
+  include(GNUInstallDirs)
+
+  set(extra_cmake_args
+    "-DCMAKE_INSTALL_LIBDIR=${CMAKE_INSTALL_LIBDIR}"
+    "-DCMAKE_INSTALL_INCLUDEDIR=${CMAKE_INSTALL_INCLUDEDIR}"
+  )
   if(DEFINED CMAKE_BUILD_TYPE)
     list(APPEND extra_cmake_args -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE})
   endif()
@@ -68,10 +73,15 @@ macro(build_console_bridge)
     DESTINATION
       ${CMAKE_INSTALL_PREFIX}
   )
+
+  set(console_bridge_INCLUDE_DIRS ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR})
+  set(console_bridge_LIBRARY_DIRS ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR})
+  set(console_bridge_LIBRARIES console_bridge)
 endmacro()
 
 if(NOT console_bridge_FOUND OR "${console_bridge_VERSION}" VERSION_LESS 1.0.1)
   build_console_bridge()
+  set(extra_test_dependencies console_bridge-1.0.1)
 endif()
 
 # TODO(blast545): This section and package.xml should be updated once copyright auto is solved.
@@ -87,8 +97,12 @@ if(BUILD_TESTING)
 
   add_performance_test(benchmark_log benchmark/benchmark_log.cpp)
   if(TARGET benchmark_log)
-    target_include_directories(benchmark_log PUBLIC include ${console_bridge_INCLUDE_DIRS})
+    target_include_directories(benchmark_log PRIVATE include ${console_bridge_INCLUDE_DIRS})
     target_link_libraries(benchmark_log ${console_bridge_LIBRARIES})
+    target_link_directories(benchmark_log PRIVATE ${console_bridge_LIBRARY_DIRS})
+    if(extra_test_dependencies)
+      add_dependencies(benchmark_log ${extra_test_dependencies})
+    endif()
   endif()
 endif()
 
```